File: ExtensionBox-main/app/src/main/java/com/extensionbox/app/Prefs.java
package com.extensionbox.app;

import android.content.Context;
import android.content.SharedPreferences;

public final class Prefs {

    private static final String FILE = "ebox";

    private static SharedPreferences p(Context c) {
        return c.getSharedPreferences(FILE, Context.MODE_PRIVATE);
    }

    // ── Module enabled ──
    public static boolean isModuleEnabled(Context c, String key, boolean def) {
        return p(c).getBoolean("m_" + key + "_enabled", def);
    }

    public static void setModuleEnabled(Context c, String key, boolean val) {
        p(c).edit().putBoolean("m_" + key + "_enabled", val).apply();
    }

    // ── Service state ──
    public static boolean isRunning(Context c) {
        return p(c).getBoolean("running", false);
    }

    public static void setRunning(Context c, boolean val) {
        p(c).edit().putBoolean("running", val).apply();
    }

    // ── Generic getters/setters for module-specific data ──
    public static int getInt(Context c, String key, int def) {
        return p(c).getInt(key, def);
    }

    public static void setInt(Context c, String key, int val) {
        p(c).edit().putInt(key, val).apply();
    }

    public static long getLong(Context c, String key, long def) {
        return p(c).getLong(key, def);
    }

    public static void setLong(Context c, String key, long val) {
        p(c).edit().putLong(key, val).apply();
    }

    public static boolean getBool(Context c, String key, boolean def) {
        return p(c).getBoolean(key, def);
    }

    public static void setBool(Context c, String key, boolean val) {
        p(c).edit().putBoolean(key, val).apply();
    }

    public static String getString(Context c, String key, String def) {
        return p(c).getString(key, def);
    }

    public static void setString(Context c, String key, String val) {
        p(c).edit().putString(key, val).apply();
    }
}

File: app/src/main/java/com/extensionbox/app/Prefs.kt
package com.extensionbox.app

import android.content.Context

object Prefs {
    private const val FILE = "ebox"

    private fun p(c: Context) = c.getSharedPreferences(FILE, Context.MODE_PRIVATE)

    fun isModuleEnabled(c: Context, key: String, def: Boolean): Boolean =
        p(c).getBoolean("m_${key}_enabled", def)

    fun setModuleEnabled(c: Context, key: String, val_: Boolean) =
        p(c).edit().putBoolean("m_${key}_enabled", val_).apply()

    fun isRunning(c: Context): Boolean = p(c).getBoolean("running", false)

    fun setRunning(c: Context, val_: Boolean) =
        p(c).edit().putBoolean("running", val_).apply()

    fun getInt(c: Context, key: String, def: Int): Int = p(c).getInt(key, def)

    fun setInt(c: Context, key: String, val_: Int) =
        p(c).edit().putInt(key, val_).apply()

    fun getLong(c: Context, key: String, def: Long): Long = p(c).getLong(key, def)

    fun setLong(c: Context, key: String, val_: Long) =
        p(c).edit().putLong(key, val_).apply()

    fun getBool(c: Context, key: String, def: Boolean): Boolean = p(c).getBoolean(key, def)

    fun setBool(c: Context, key: String, val_: Boolean) =
        p(c).edit().putBoolean(key, val_).apply()

    fun getString(c: Context, key: String, def: String?): String? = p(c).getString(key, def)

    fun setString(c: Context, key: String, val_: String?) =
        p(c).edit().putString(key, val_).apply()

    fun getAll(c: Context): Map<String, *> = p(c).all

    fun clearAll(c: Context) = p(c).edit().clear().apply()

    fun resetDailyStats(c: Context) {
        p(c).edit().apply {
            putInt("ulk_today", 0)
            putLong("stp_today", 0L)
            putLong("dat_daily_total", 0L)
            putLong("dat_daily_wifi", 0L)
            putLong("dat_daily_mobile", 0L)
            putLong("scr_on_acc", 0L)
            putInt("fap_today", 0)
            apply()
        }
    }

    fun importJson(c: Context, json: String) {
        try {
            val jsonObject = org.json.JSONObject(json)
            val editor = p(c).edit()
            val iterator = jsonObject.keys()
            while (iterator.hasNext()) {
                val key = iterator.next()
                when (val value = jsonObject.get(key)) {
                    is Boolean -> editor.putBoolean(key, value)
                    is Int -> editor.putInt(key, value)
                    is Long -> editor.putLong(key, value)
                    is Float -> editor.putFloat(key, value)
                    is String -> editor.putString(key, value)
                    else -> editor.putString(key, value.toString())
                }
            }
            editor.apply()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
